src/Makefile

```diff
@@ -8,18 +8,24 @@
 CXX=g++
 C=gcc
 
+export CXXFLAGS
+export C
+export CC
+export CXX
+export LDFLAGS
+export LIBFLAGS
+
 # Compiler flags
 CFLAGS=-O3 -D_FILE_OFFSET_BITS=64 -g
```

Comment:
This should read

CFLAGS += -O3 -D_FILE_OFFSET_BITS=64 -g

or even better

CFLAGS += -D_FILE_OFFSET_BITS=64

You could have tested for the variable being empty and only dare to ovewrite/set it via

CFLAGS += -O3 -D_FILE_OFFSET_BITS=64 -g

Anyway it is unfair to force aggressive -O3 optimization. Leave it up to the user. If you read the gcc manual you will find that -O3 does not guarantee anymore the best speed. Go for -O2 if you want to stay on teh safe side. More meaningful would have been an attempt to add -march=tune. But this should be done by configure and not hardcoded in a Makefile. The above are all wrong approaches.

Ideally, use configure to detect type of compiler (-O3 has different meaning with other compilers) and supported CFLAGS/CXXGLAGS. Respect user's setting so only append to existing `CC`, `CXX`, `CFLAGS` and `CXXFLAGS` variables using the `+=` way.